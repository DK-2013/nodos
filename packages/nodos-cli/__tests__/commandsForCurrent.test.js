import path from 'path';
import { nodos } from '@nodosjs/core';
import runCurrent from '../lib/current.js';

const projectRoot = path.join(__dirname, '../__fixtures__/application');

test('nodos/console', async () => {
  const replServer = { context: {} };
  const container = {
    repl: {
      start: () => replServer,
    },
  };

  const app = nodos(projectRoot);
  await runCurrent(app, { container, args: ['console'] });
  expect(replServer.context).toHaveProperty('app');
});

// NOTE jest somehow doesn't wait for all async functions to finish and calls check BEFORE mocked function call
// That's why currently we just need to put check after function call just by placing in later in event loop
test('nodos/server', async () => {
  const app = nodos(projectRoot);
  app.listen = jest.fn().mockResolvedValue(42);
  await runCurrent(app, { args: ['server'] });
  setTimeout(() => {
    expect(app.listen).toHaveBeenCalled();
  }, 0);
});

test('nodos/routes', async () => {
  const app = nodos(projectRoot);
  const container = {
    print: (output) => { expect(output).toMatchSnapshot(); },
  };
  await runCurrent(app, { container, args: ['routes'] });
});
