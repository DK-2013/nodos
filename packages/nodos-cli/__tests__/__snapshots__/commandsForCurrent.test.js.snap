// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodos/routes 1`] = `
"Name Verb URI Pattern Pipeline Controller#Action
root GET                       home#index       "
`;
